File: 20230413/js/file.js
const fs = require("fs");

module.exports = class File{
    constructor(){
        this.filesList = {};
        this.access = {};
        this.accessIDCount = 0;
    }
    
    newFile(_file = null, _keys = []){
        let currentFile = null;
        let currentObject = null;
        let id = this.accessIDCount;

        if (this.filesList[_file]) currentFile = this.filesList[_file];

        else{
            try { require(_file) }
            catch(e) { throw  `No Such A File : ${_file}` }
            finally { currentFile = require(_file) }

            this.filesList[_file] = currentFile;
        }

        currentObject = currentFile;

        for (let key of _keys){
            if (!currentObject[key]) throw `Unaccessible : ${key}\nOf ${_file}\n${_keys}`;

            currentObject = currentObject[key];
        }

        this.access[id] = {
            "id" : id,
            "file" : _file,
            "keys" : _keys
        };

        this.accessIDCount++;

        return id;
    }

    editFile(id, _keys = [], value){
        if (!_keys) return;
        if (!this.access[id]) throw `No Such an ID : ${id}`;

        const _file = this.access[id].file;
        const keys = this.access[id].keys.concat(_keys);

        let currentObject = null;
        let WholeFile = null;

        WholeFile = this.filesList[_file];
        currentObject = WholeFile;

        for (let i = 0; i < keys.length - 1; i++){
            if (!currentObject[keys[i]]) throw `Unaccessible : ${keys[i]}\nOf ${_file}\n${keys}`;

            currentObject = currentObject[keys[i]];
        }

        currentObject[keys[keys.length - 1]] = value;

        this.filesList[_file] = WholeFile;

        this.applyFile.bind(this)(_file);
    }

    applyFile(_file){
        let updatedJson = JSON.stringify(this.filesList[_file], null, 4);

        fs.writeFile(_file, updatedJson, "utf8", (err) => {
            if (err) throw `Having a Problem Accessing To : ${_file}\n${err}`;
        });
    }
}

File: 20230413/C#/file.cs
using System;
using System.IO;
using System.Collections.Generic;
using Newtonsoft.Json;

public class File
{
    private Dictionary<string, dynamic> filesList;
    private Dictionary<int, Dictionary<string, dynamic>> access;
    private int accessIDCount;

    public File()
    {
        this.filesList = new Dictionary<string, dynamic>();
        this.access = new Dictionary<int, Dictionary<string, dynamic>>();
        this.accessIDCount = 0;
    }

    public int NewFile(string file = null, List<string> keys = null){
        string currentFile = null;
        dynamic currentObject = null;
        int id = this.accessIDCount;

        if (this.filesList.ContainsKey(file))
        {
            currentFile = this.filesList[file];
        }
        else
        {
            try
            {
                File.ReadAllText(file);
            }
            catch (Exception e)
            {
                throw new Exception($"No Such A File : {file}");
            }
            finally
            {
                currentFile = File.ReadAllText(file);
            }

            this.filesList[file] = currentFile;
        }

        currentObject = JsonConvert.DeserializeObject<dynamic>(currentFile);

        if (keys != null)
        {
            foreach (string key in keys)
            {
                if (currentObject[key] == null) throw new Exception($"Unaccessible : {key}\nOf {file}\n{keys}");

                currentObject = currentObject[key];
            }
        }

        this.access[id] = new Dictionary<string, dynamic>
        {
            { "id", id },
            { "file", file },
            { "keys", keys }
        };

        this.accessIDCount++;

        return id;
    }

    public void EditFile(int id, List<string> keys = null, dynamic value = null)
    {
        if (keys == null) return;
        if (!this.access.ContainsKey(id)) throw new Exception($"No Such an ID : {id}");

        string file = this.access[id]["file"];
        List<string> accessKeys = this.access[id]["keys"];

        if (keys != null) accessKeys.AddRange(keys);

        dynamic currentObject = null;
        dynamic wholeFile = null;

        wholeFile = JsonConvert.DeserializeObject<dynamic>(this.filesList[file]);
        currentObject = wholeFile;

        for (int i = 0; i < accessKeys.Count - 1; i++)
        {
            if (currentObject[accessKeys[i]] == null) throw new Exception($"Unaccessible : {accessKeys[i]}\nOf {file}\n{accessKeys}");

            currentObject = currentObject[accessKeys[i]];
        }

        currentObject[accessKeys[accessKeys.Count - 1]] = value;

        this.filesList[file] = JsonConvert.SerializeObject(wholeFile);

        this.ApplyFile(file);
    }

    public void ApplyFile(string file)
    {
        string updatedJson = JsonConvert.SerializeObject(this.filesList[file]);

        try
        {
            File.WriteAllText(file, updatedJson);
        }
        catch (Exception e)
        {
            throw new Exception($"Having a Problem Accessing To : {file}\n{e}");
        }
    }
}
